Source/DafnyRuntime/DafnyRuntimeJava/src/main/java/dafny/CodePoint.java

```diff
@@ -0,0 +1,62 @@
+package dafny;
+
+/**
+ * An int wrapper type just like java.lang.Integer,
+ * but used as a more type-safe reference to a Unicode scalar value
+ * specifically, which corresponds to the Dafny `char` type
+ * when using --unicode-char.
+ */
+public final class CodePoint {
+
+    // Caching a subset of values just like the built-in box types like Character.
+    private static class CodePointCache {
+        private CodePointCache(){}
+
+        static final CodePoint cache[] = new CodePoint[127 + 1];
```

Comment:
I assume there's some reason to think of this size as `127 + 1` (instead of just `128`), but I don't understand what it is. I suggest you add a comment.

Reply:
Not a good reason :) It's copied almost verbatim from `java.lang.Character`, and the built-in boxing types are a little more general where the limit is configurable for some types, and some handle values less than 0. I've cleaned this up now.